tests/unit/test_charm.py

```diff
@@ -479,3 +479,30 @@ def test_given_user_provided_certificates_and_unit_is_leader_when_config_changed
         self.harness.update_config(key_values=key_values)
 
         patch_revoke_all_certificates.assert_called_with()
+
+    @patch("charm.generate_certificate")
+    def test_generate_certificate_action(self, patch_generate_certificate):
```

Comment:
Please follow the same test naming convention as the other tests here.

Reply:
Done